File: Segundo/PDOO/ruby p4/SpaceCity.rb
require_relative'SpaceStation'
require_relative'Transformation'

module Deepspace
    class SpaceCity < SpaceStation
        def initialize(base,rest)
            #cambiar a super(blabla)
            @ammoPower=base.ammoPower
            @fuelUnits=base.fuelUnits
            @name=base.name
            @nMedals=base.nMedals
            @shieldPower=base.shieldPower
            @pendingDamage=base.pendingDamage
            @weapons=base.weapons
            @shieldBoosters=base.shieldBooster
            @hangar=base.hangar

            @collaborators=ArrayList.new
            count = rest.length
            while(count>0)
                @rest[count]=rest[count]
                count-=1
            end
        end

        def collaborators
            return @collaborators
        end

        def fire
            suma = super
            count = 0
            while(count < @rest.length)
                suma+=@rest[count].fire
                count+=1
            end
            return suma
        end

        def protection
            suma=super
            count = 0
            while(count < @rest.length)
                suma+=@rest[count].protection
                count+=1
            end
            return suma
        end

        def setLoot(loot)
            #not sure
            super
            return Transformation::NOTRANSFORM
        end
            
    end
end

File: Segundo/PDOO/deepSpace/src/deepspace/SpaceCity.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;
import java.util.ArrayList;

/**
 *
 * @author carmenc
 */
public class SpaceCity extends SpaceStation{
    ArrayList<SpaceStation> collaborators = new ArrayList<>();
    
    public SpaceCity(SpaceStation base, ArrayList<SpaceStation> rest){
        super(base);
        collaborators=(ArrayList<SpaceStation>) rest.clone();
    }
    
    @Override
    public float fire(){
        float p=super.fire();
        for(int i = 0; i < collaborators.size(); i++){
            p+=collaborators.get(i).fire();
        }
        return p;
    }
    
    public ArrayList<SpaceStation> getCollaborators(){
        return collaborators;
    }
    
    @Override
    public SpaceStationToUI getUIversion(){
        return new SpaceCityToUI(this);
    }
    
    @Override
    public Transformation setLoot(Loot loot){
        return Transformation.NOTRANSFORM;
    }
    
    @Override
    public float protection(){
        float p=super.protection();
        for(int i = 0; i < collaborators.size(); i++){
            p+=collaborators.get(i).protection();
        }
        return p;
    }
    
//    @Override
//    public ShotResult receiveShot(float shot){
//        
//    }
}
